v0/destinations/splitio/transform.js

```diff
@@ -0,0 +1,108 @@
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  KEY_CHECK_LIST,
+  EVENT_TYPE_ID_REGEX
+} = require("./config");
+const { EventType } = require("../../../constants");
+const {
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  constructPayload,
+  flattenJson,
+  isDefinedAndNotNullAndNotEmpty,
+  getFieldValueFromMessage
+} = require("../../util");
+
+function responseBuilderSimple(payload, category, destination) {
+  if (payload) {
+    const responseBody = payload;
+    const response = defaultRequestConfig();
+    response.endpoint = category.endPoint;
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${destination.Config.apiKey}`
+    };
+    response.body.JSON = removeUndefinedAndNullValues(responseBody);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new Error("Payload could not be constructed");
+}
+
+function populateOutputProperty(inputObject) {
+  const outputProperty = {};
+  Object.keys(inputObject).forEach(key => {
+    if (!KEY_CHECK_LIST.includes(key) && !Array.isArray(inputObject[key])) {
+      outputProperty[key] = inputObject[key];
+    }
+  });
+  return outputProperty;
+}
+
+function prepareResponse(message, destination, category) {
+  let bufferProperty = {};
+  const { environment, trafficType } = destination.Config;
+  const { type } = message;
+
+  let outputPayload = {};
+
+  outputPayload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  outputPayload.eventTypeId = outputPayload.eventTypeId.replace(/ /g, "_");
+  if (EVENT_TYPE_ID_REGEX.test(outputPayload.eventTypeId)) {
+    switch (type) {
+      case EventType.IDENTIFY:
+        bufferProperty = populateOutputProperty(
+          getFieldValueFromMessage(message, "traits")
```

Comment:
getFieldValueFromMessage can return null. We're not handling null in populateOutputProperty.

I'd suggest to have the checking in the method and remove from outside. It's better for readability as well - you're validating before accessing it

Reply:
Also, lets add test cases and for these